src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -190,6 +211,18 @@ void setupParser() throws IOException, AsyncApiException {
     resultIdIndex++;
   }
 
+  public InputStream getQueryResultStream(BulkConnection bulkConnection) throws SalesforceQueryExecutionException {
+    try {
+      return bulkConnection.getQueryResultStream(jobId, batchId, resultIds[resultIdIndex]);
+    } catch (AsyncApiException exception) {
+      LOG.error("The bulk query job {} failed.", jobId);
+      if (RETRY_ON_REASON.contains(exception.getExceptionCode())) {
+        throw new SalesforceQueryExecutionException(exception.getMessage());
+      }
```

Comment:
if the error is not retryable, we are ignoring the error here, is it expected?

Reply:
error logged